packages/rs-drive-abci/src/execution/validation/state_transition/processor/v0/mod.rs

```diff
@@ -549,133 +583,117 @@ impl StateTransitionAdvancedStructureValidationV0 for StateTransition {
+    }
 
-                        Ok(validation_result)
-                    }
-                    None => Err(Error::Execution(ExecutionError::VersionNotActive {
-                        method: "identity create transition: validate_identity_and_signatures"
-                            .to_string(),
-                        known_versions: vec![0],
-                    })),
-                    Some(version) => {
-                        Err(Error::Execution(ExecutionError::UnknownVersionMismatch {
-                            method: "identity create transition: validate_identity_and_signatures"
-                                .to_string(),
-                            known_versions: vec![0],
-                            received: version,
-                        }))
-                    }
-                }
-            }
+    fn retrieve_identity_info(
+        &self,
+        drive: &Drive,
+        tx: TransactionArg,
+        _execution_context: &mut StateTransitionExecutionContext,
+        platform_version: &PlatformVersion,
+    ) -> Result<ConsensusValidationResult<PartialIdentity>, Error> {
+        match self {
             StateTransition::IdentityTopUp(st) => {
-                match platform_version
-                    .drive_abci
-                    .validation_and_processing
-                    .state_transitions
-                    .identity_top_up_state_transition
-                    .identity_signatures
-                {
-                    // The validation of the signature happens on the state level
-                    Some(0) => Ok(st
-                        .retrieve_topped_up_identity(drive, tx, platform_version)?
-                        .map(Some)),
-                    None => Err(Error::Execution(ExecutionError::VersionNotActive {
-                        method: "identity top up transition: validate_identity_and_signatures"
-                            .to_string(),
-                        known_versions: vec![0],
-                    })),
-                    Some(version) => {
-                        Err(Error::Execution(ExecutionError::UnknownVersionMismatch {
-                            method: "identity top up transition: validate_identity_and_signatures"
-                                .to_string(),
-                            known_versions: vec![0],
-                            received: version,
-                        }))
-                    }
-                }
+                Ok(st.retrieve_topped_up_identity(drive, tx, platform_version)?)
             }
+            _ => Ok(ConsensusValidationResult::new()),
         }
     }
+
+    /// Is the state transition supposed to have an identity in the state to succeed
+    fn uses_identity_in_state(&self) -> bool {
+        !matches!(self, StateTransition::IdentityCreate(_))
+    }
+
+    /// Do we validate the signature based on identity info?
+    fn validates_signature_based_on_identity_info(&self) -> bool {
+        !matches!(
```

Comment:
I don't think we need to, as this does not error. We are probably never going to have have asset lock state transitions other that identity top up and identity create.

Reply:
I mean there are many other similar functions there. Using matches! is not safe because it won't show us if we did't add this ST where we supposed to. With `match` we gonna have compilation errors.

Reply:
You fixed this